### DISCUSSION TITLE
Is mitmproxy detectable?

### DISCUSSION THREAD

**Discussion by DnyyGzd** [mitmproxy/mitmproxy#6055] [General]
I am using the default mode of mitmproxy. I have a question, is it possible for a web server to detect the use of the mitmproxy?

**Comment by mhils** [comment]
mitmproxy will pass through HTTP requests as-is (so typically no way to detect it on this layer). However, you could definitely fingerprint the TLS ClientHello, which will be different.

**Reply by Prinzhorn** [reply]
https://github.com/mitmproxy/mitmproxy/issues/4575

**Reply by DnyyGzd** [reply]
I've read thread #4575, is there a way to use a browser-like TLS ClientHello?

**Comment by mhils** [comment]
Not right now.
